src/spec.rs

```diff
@@ -99,18 +99,21 @@ impl FromStr for Endpoints {
                 let span = tracing::error_span!("parse_addr", ?addr);
                 let _g = span.enter();
 
-                // Addresses may have the suffix `#h2` to indicate they support h2 upgrading.
-                let mut parts = addr.splitn(2, '#');
-                match (parts.next(), parts.next()) {
-                    (Some(addr), h2) => match addr.parse() {
+                // Endpoints can be configured for h2 upgrading and identity with the '#' suffix:
+                // - `#h2` supports h2 upgrading
+                // - `#h2#x-identity` supports h2 upgrading and has the `x-identity` TLS identity
+                // - `##x-identity` has the `x-identity` TLS identity
```

Comment:
Can we also add this to the `structopt` doc comments in `main.rs`, here? https://github.com/linkerd/linkerd2-mock-dst/blob/4a8cd02c36e2a15b41f30260db1bd45eb84eac18/src/main.rs#L22-L23

That way, this will show up in the CLI help output.

Also, it was not super clear to me from the comment whether `x-identity` was part of the syntax, or was a placeholder for the name of the endpoint's identity — I had to read the code to figure out what that meant. Maybe we should change this to something like
```suggestion
                // Endpoints can be configured for h2 upgrading and identity with the '#' suffix:
                // - `#h2` supports h2 upgrading
                // - `#h2#<IDENTITY>` supports h2 upgrading and has the `<IDENTITY>` TLS identity
                // - `##<IDENTITY>` has the `<IDENTITY>` TLS identity
```
to maybe make it a little clearer that it's a placeholder?